benchmark_runner/common/grafana/update_grafana_latest_value_mappings.py

```diff
@@ -45,7 +45,16 @@ def get_last_elasticsearch_versions(self, last_es_fetch_days=LAST_ES_FETCH_DAYS)
             for version, value in data['_source'].items():
                 if version in display_versions and value not in new_versions.values() and len(value) < self.MAX_VERSION_LEN:
                     # Display only numbers in the Grafana panel, removing characters
-                    new_versions[value.translate(str.maketrans('', '', '.-rcef'))] = value
+                    if '-ec.' in value:
+                        full_version = value
+                        value = ''.join('0' if c == 'e' else c for c in value if c not in '.cfr-')
+                        new_versions[value] = full_version
+                    elif '-rc.' in value:
+                        full_version = value
+                        value = ''.join('1' if c == 'r' else c for c in value if c not in '.cfe-')
+                        new_versions[value] = full_version
+                    else:
+                        new_versions[value.translate(str.maketrans('', '', '.-rcef'))] = value
```

Comment:
This code is duplicated in two places.  Please create a single function that encapsulates both, and simplify the logic.  You don't need that for loop here at all.

Reply:
I have have merged it to 1 method.